### PR TITLE
Add `FORCE_COLOR=1` to default environment variables

### PR DESCRIPTION
**Pull request by Alhadis**
To cut a long story short, Mocha's coloured output is [causing Emacs users](https://emacs.stackexchange.com/questions/27218/why-is-there-so-often-ugly-output-in-node-js-compilation-buffers) grief because Emac's default `compile` command chokes on ANSI escape sequences when emitted from `make -k` feedback. @plroebuck identified [the crux](https://github.com/chalk/supports-color/issues/88) of the problem  in [`chalk/supports-color`](https://github.com/chalk/supports-color) handling of `TERM=dumb` environments.

I submitted [a pull-request](https://github.com/chalk/supports-color/pull/89) which fixes this, and simultaneously fixes `compiler-mode`'s problems whilst affecting the coloured output of `mocha.el`:

Mocha.el: Before Mocha.el: After

Basically, the fix is to add `FORCE_COLOR=1` in the environment, which is picked up by the `supports-color` module as a user's wish to go and use coloured output anyway, irrespective o f the `TERM=dumb` variable being set in both compiler-modes.

We can merge this now, it won't get noticed because the package's highlighting is currently working fine. But users will notice once `chalk/supports-color#89` gets merged and shipped in a new release;  having this the default for `mocha-environment-variables` ensures there's absolutely no friction when it does. =)

### CODE REVIEW

**Comment by plroebuck** [comment]
The "supports-color" PR for `TERM=dumb` does the same thing as `FORCE_COLOR="0"` would.
Wouldn't this technically be undoing the effect of chalk/supports-color#89 (when implemented)?

**Comment by Alhadis** [comment]
Nope. :=) Like I said in another thread, @scottaj gets it right because the escapes are going where they have to. But the command that was screwing it up was launching an extra shell per process, which means the ANSI escapes were interpreted literally by their <a href="https://e.vampirefreaks.com/emotes/quote.gif"?>dumb</ href="https://e.vampirefreaks.com/emotes/quote.gif"> terminals before being piped through to the top-level process layer or something....

**Comment by Alhadis** [comment]
So we're not undoing anything because we're giving his application  a chance to force colours even if the `TERM=dumb` was already being set in his environment each time.

**Comment by Alhadis** [comment]
Nudging @scottaj for review.

**Comment by Alhadis** [comment]
@scottaj Err... little help?
